src/harness/testcases/WINNF_FT_S_MCP_testcase.py

```diff
@@ -306,3 +315,227 @@ def test_WINNF_FT_S_MCP_1(self, config_filename):
+
+        # check MCP.1
+        # TODO: iap_refmodel_margin_list input required
+        #self.checkMcpIap(iteration_content['protectedEntities'], agg_interference_list, iap_refmodel_margin_list, config['deltaIap'])
+
+      # Step 13 : Configure SAS Harness with FAD,trigger FAD generation
+      for index, test_harness in enumerate(sas_test_harness_objects):
+        sas_test_harness_dump_records = [iteration_content['sasTestHarnessData'][index]['cbsdRecords']]
+        test_harness.writeFadRecords(sas_test_harness_dump_records)
+
+      # Step 14,15 : Trigger Full Activty Dump and Pull FAD records from SAS UUT
+      fad_uut_object = getFullActivityDumpSasUut(self._sas,
+                                                 self._sas_admin,
+                                                 config['domainProxyConfigs'][0]['cert'],
+                                                 config['domainProxyConfigs'][0]['key'])
+
+      # Step 16 : Trigger CPAS and wait for its completion
+      self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()
+
+      # Step 17 : Call IAP reference model
+      # TODO: To invoke DPA Move Reference model
+
+      # Call Pre IAP model
+      # TODO: To invoke Pre-iap reference model
+      #pre_iap.pre_iap(protected_entities, fad_uut_object, fad_test_harnesses_objects, pre_iap_headRooms)
+
+      # Call IAP reference model
+      # TODO: To invoke IAP reference model
+      # iap_refmodel_margin_list = iap.performIap(entities, fad_uut_object, fad_test_harnesses_objects, pal_records)
+
+      # Step 18,19,20 and 21 :
+      # Send heartbeat request for the grants, relinquish the grant, grant request and heartbeat for new grant
+      for domain_proxy_object in domain_proxy_objects:
+        domain_proxy_object.performHeartbeatAndUpdateGrants()
+
+        # Step 22: Calculating the Aggregate interface and IAP reference model invoke.
+        cbsd_agg_list.append(domain_proxy_object.getCbsdsWithAtLeastOneAuthorizedGrant())
+
+      # Invoke Aggregate Interference Model
+      # TODO: To invoke Aggregate Interference Model
+      #for protected_entity in iteration_content['protectedEntities'].keys():
+      #  agg_interference_list = self.callAggregateInterferenceModel(cbsd_agg_list, iteration_content['protectedEntities'])
+
+        # Check MCP.1
+        #self.checkMcpIap(iteration_content['protectedEntities'], agg_interference_list, iap_refmodel_margin_list, config['deltaIap'])
+
+      # Step 23: ESC Test harness enables DPA activations
+      # deactivated the dpaDeactivationList DPA IDs
+      for dpa in iteration_content['dpaActivationList']:
+        self._sas_admin.TriggerDpaDeactivation(dpa)
+
+      # activated the dpaActivationList DPA IDs
+      # step 24: wait for 240 sec if DPA is activated in step 23 else 15 sec
+      for dpa in iteration_content['dpaActivationList']:
+        self._sas_admin.TriggerDpaActivation(dpa)
+        time.sleep(240)
```

Comment:
You sleep once after all of them, not once each.

Reply:
Updated. Sleep is invoked after the for loop with condition.